pylib/zeus/zeus.py

```diff
@@ -30,15 +30,20 @@ def find_zeus_executable(current_path):
   :return: Path to the executable or None if not found.
   """
 
+  wanted_steps = (None, None)
   while True:
+    wanted_steps = (wanted_steps[1], os.path.basename(os.path.normpath(current_path)))
     # Check all files in the current directory
-    for file in os.listdir(current_path):
-      # Check if the file starts with 'zeus_' and has no extension
-      if file.startswith('zeus_') and '.' not in file:
-        file_path = os.path.join(current_path, file)
-        # Check if the file is executable
-        if os.access(file_path, os.X_OK):
-          return file_path
+    list_dir = frozenset(os.listdir(current_path))
+    for wanted_step in wanted_steps:
+        if wanted_step is None:
+            continue
+        file_name = 'zeus_' + wanted_step
```

Comment:
Crazy edge case - but if I'm in `generate/generate.sh` then it will look for `zeus_generate.sh` which again contradicts the comment and is a probably over pedantic comment.

Reply:
yeah unfortunately to make it ideal we need to refactor the whole script and remove these copy-pasted executables whatsoever. I just try to make it better with what we have

Reply:
That's fine - it's a very unlikely edge case.